pkg/flowaggregator/flowaggregator.go

```diff
@@ -201,7 +276,25 @@ func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
 					continue
 				}
 			}
-			err := fa.aggregationProcess.ForAllRecordsDo(fa.sendFlowKeyRecords)
+			dataSet := ipfix.NewSet(ipfixentities.Data, fa.templateID, false)
+			addFlowKeyRecord := func(key ipfixintermediate.FlowKey, record ipfixintermediate.AggregationFlowRecord) error {
+				if !record.ReadyToSend {
+					klog.V(4).Info("Skip sending record that is not correlated.")
+					return nil
+				}
+				// if current message size is close to size limit, send the dataSet before adding the new record
+				if dataSet.GetSet().GetBuffLen()+msgHeaderLen > fa.exportingProcess.GetMsgSizeLimit()*3/4 {
+					_, err = fa.sendDataSet(dataSet)
+					if err != nil {
+						return err
+					}
+					fa.sentFlowKeys = append(fa.sentFlowKeys, &key)
```

Comment:
this is also modified in yongming's PR. Please rebase when you get a chance

Reply:
Sure. Thanks!